gwpy/cli/cliproduct.py

```diff
@@ -801,13 +803,14 @@ def annotate_save_plot(self, args):
         all_units = set()
         for ts in self.timeseries:
             un = str(ts.unit)
-            if (un != 'undef') & (un != ''):
+            if (un != 'undef') & (un != '') & (un != '*'):
```

Comment:
it might be cleaner to set up a list in the module header that contains all of the proscribed units,

```python
BAD_UNITS = {'undef', '', '*'}
```

then just do something like

```python
            if str(ts.unit) in BAD_UNITS:
                ts.override_unit('')
            else:
                all_units.add(str(ts.unit))

Reply:
I've been wondering whether they should be handled at this level at all? It was originally put in there to avoid confusion with the transition. The only real problem is when we do arithmetic on the units and for the most part that is being handled in the timeseries object I believe.

I like your suggestion and I will make the change.

Reply:
@areeda, the units parser knows about `'undef'`, and `''` is a legitimate way of specifying 'dimensionless', so please only include `'*'` in your `BAD_UNITS` `set`. We might consider handling that at the unit parser level, as you say.